### PR TITLE
Jaxrs21tck issues

### PR DESCRIPTION
**Pull request by andymc12**
Two issues:
1. Invocation.Builder.headers(...) should replace/remove all pre-existing headers.
2. An empty, but non-null country portion of a Locale should be treated similar to a null country - i.e. new Locale("en", "") should result in "en", not "en-".

### CODE REVIEW

**Comment by andymc12** [comment]
#build

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/spec/InvocationBuilderImpl.java [hunk 1]
```diff
@@ -257,6 +257,7 @@ public Builder header(String name, Object value) {
 
     @Override
     public Builder headers(MultivaluedMap<String, Object> headers) {
+        webClient.removeAllHeaders();
         RuntimeDelegate rd = HttpUtils.getOtherRuntimeDelegate();
         for (Map.Entry<String, List<Object>> entry : headers.entrySet()) {
```
Comment (hibell) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/spec/InvocationBuilderImpl.java]: The API for Invocation.Builder#headers() states that passing a null value in for headers needs to remove all existing headers:
https://jax-rs.github.io/apidocs/2.1/javax/ws/rs/client/Invocation.Builder.html#headers-javax.ws.rs.core.MultivaluedMap-

At line 262, passing in null will cause an NPE when headers.entrySet() is invoked so a null check will also be necessary.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/AbstractClient.java [hunk 2]
```diff
@@ -375,6 +375,10 @@ public void close() {
         closed = null;
     }
 
+    public void removeAllHeaders() {
```
Comment (hibell) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/AbstractClient.java]: This looks to be a duplicate of removeAllHeaders() on [L175](https://github.com/andymc12/open-liberty/blob/f4a9461693694e01b31feb3c3ad4adf5e5e6b894/dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/AbstractClient.java#L175).

**Comment by andymc12** [comment]
@jkoehler22 @hibell thanks for the review!  Updates are in. New #build

**Comment by andymc12** [comment]
#build

**Pull request merged**